javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/PropagatorsInitializer.java

```diff
@@ -39,9 +42,25 @@
           .put(JAEGER, JaegerPropagator.getInstance())
           .put(OT_TRACER, OtTracerPropagator.getInstance())
           .put(XRAY, AwsXRayPropagator.getInstance())
+          .put(BAGGAGE, W3CBaggagePropagator.getInstance())
           .build();
 
-  /** Initialize OpenTelemetry global Propagators with propagator list, if any. */
+  /**
+   * Initialize OpenTelemetry global Propagators with propagator list, if any.
+   *
+   * <p>Because TraceMultiPropagator returns first successful extracted Context and stops further
+   * extraction next rules applied:
+   *
+   * <ul>
+   *   <li>W3CBaggagePropagator is added to DefaultContextPropagators to allow another propagator
+   *       extract Context.
+   *   <li>JaegerPropagator is added to DefaultContextPropagators because it extracts both Context
+   *       and Baggage. Otherwise in TraceMultiPropagator it may not get a chance to extract any
+   *       existing Baggage.
+   *   <li>W3CBaggagePropagator comes after JaegerPropagator, as it can have more complex/complete
```

Comment:
Above you say "returns first successful extracted Context", here you seem to say that later Baggage extraction can enrich the previous one. If context and baggage extractions behave differently in this regard, we have to say so explicitly.

Reply:
updated.